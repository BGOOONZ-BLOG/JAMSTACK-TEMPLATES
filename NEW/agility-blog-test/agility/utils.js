const renderHTML = (html) => {
	if (!html) return { __html: "" };
	return { __html: cleanHTML(html) };
}

const cleanHTML = (html) => {
	if (!html) return ""

	//fix '~' in links in HTML
	return html.replace(/href="~\//gi, 'href="/')
}

const asyncForEach = async (array, callback) => {
	for (let index = 0; index < array.length; index++) {
		await callback(array[index], index, array);
	}
}


const expandContentList = async ({ agility, contentItems, languageCode, depth }) => {

	asyncForEach(contentItems, async (contentItem) => {
		await expandContentItem({agility, contentItem, languageCode, depth})
	})

}

const expandContentItem = async ({ agility, contentItem, languageCode, depth = 1 }) => {
	if (!contentItem) return null;



	const api = agility;

	if (depth > 0) {
		//make this work for the .fields or the .customFields property...
		let fields = contentItem.fields;
		if (!fields) fields = contentItem.customFields;

		for (const fieldName in fields) {
			const fieldValue = fields[fieldName];

			if (fieldValue.contentid > 0) {
				//single linked item
				const childItem = await api.getContentItem({ contentID: fieldValue.contentid, languageCode, depth: depth - 1 });
				if (childItem != null) fields[fieldName] = childItem;
			} else if (fieldValue.sortids && fieldValue.sortids.split) {
				//multi linked item
				const sortIDAry = fieldValue.sortids.split(',');
				const childItems = [];
				for (const childItemID of sortIDAry) {
					const childItem = await api.getContentItem({ contentID: childItemID, languageCode, depth: depth - 1 });
					if (childItem != null) childItems.push(childItem);
				}
				fields[fieldName] = childItems;
			}
		}
	}
	return contentItem;
}

const expandLinkedList = async ({ agility, contentItem, languageCode, fieldName, sortIDField  }) => {
	if (!contentItem) return null;

	let fieldObj = contentItem.fields[fieldName]
	if (! fieldObj) throw Error(`The field ${fieldName} was not found on the content item.`)

	const referenceName = fieldObj.referencename
	if (! referenceName) throw Error(`A referencename property was not found on the ${fieldName} value.`)

	//grab the content list...
	let listItems = await agility.getContentList({referenceName, languageCode})
	if (listItems?.length > 0) {
		let sortIDs = contentItem.fields[sortIDField]
		if (sortIDs?.length > 0 && sortIDs?.split) {
			//if we have sort ids, assemble the values in the list based on those...
			const sortIDAry = sortIDs.split(',');
			const sortedItems = [];
			for (const idStr of sortIDAry) {
				const childContentID = parseInt(idStr)

				const childItemIndex = listItems.findIndex(item => item.contentID === childContentID)
				if (childItemIndex >= 0) {
					sortedItems.push(listItems[childItemIndex]);
					listItems.splice(childItemIndex, 1)

				}
			}

			listItems = sortedItems.concat(listItems)
		}
	}

	contentItem.fields[fieldName] = listItems;
	return contentItem;

}



module.exports = {
	renderHTML,
	cleanHTML,
	asyncForEach,
	expandContentItem,
	expandContentList,
	expandLinkedList
}