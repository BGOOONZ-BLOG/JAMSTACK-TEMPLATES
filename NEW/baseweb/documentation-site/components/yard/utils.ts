import {TProp} from 'react-view';
import {TProviderValue} from './provider';

export type TPropValueOverrides = {
  [key: string]: {
    active: boolean;
    style: string;
  };
};

export const countProps = (
  props: {[key: string]: TProp},
  propsConfig: {[key: string]: TProp},
) => {
  let changedProps = 0;
  Object.keys(props).forEach(prop => {
    if (
      prop !== 'overrides' &&
      props[prop].value !== '' &&
      typeof props[prop].value !== 'undefined' &&
      //@ts-ignore
      props[prop].value !== propsConfig[prop].value
    ) {
      changedProps++;
    }
  });
  return changedProps;
};

export const countOverrides = (overrides: any) => {
  if (!overrides) {
    return 0;
  }
  const existingOverrides = overrides.value ? Object.keys(overrides.value) : [];
  return existingOverrides.filter(key => overrides.value[key].active).length;
};

export const countThemeValues = (themeState: TProviderValue) => {
  if (!themeState) return 0;
  return Object.keys(themeState).length;
};
