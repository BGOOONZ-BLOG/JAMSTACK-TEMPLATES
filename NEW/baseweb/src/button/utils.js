/*
Copyright (c) 2018-2020 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import type {ButtonPropsT} from './types.js';

export function getSharedProps({
  disabled,
  isLoading,
  isSelected,
  kind,
  shape,
  size,
}: ButtonPropsT) {
  return {
    $disabled: disabled,
    $isLoading: isLoading,
    $isSelected: isSelected,
    $kind: kind,
    $shape: shape,
    $size: size,
  };
}
