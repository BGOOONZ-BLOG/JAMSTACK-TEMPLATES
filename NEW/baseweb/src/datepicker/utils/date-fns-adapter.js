/*
Copyright (c) 2018-2020 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import DateFnsUtils from '@date-io/date-fns';
import type {DateIOAdapter} from './types.js';

const adapter: DateIOAdapter<Date> = new DateFnsUtils({});

export default adapter;
