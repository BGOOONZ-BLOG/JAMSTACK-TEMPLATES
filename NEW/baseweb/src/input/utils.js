/*
Copyright (c) 2018-2020 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import type {InputPropsT, BaseInputPropsT, InternalStateT} from './types.js';

export function getSharedProps<T>(
  props: BaseInputPropsT<T> | InputPropsT,
  state: InternalStateT,
) {
  const {disabled, error, positive, adjoined, size, required} = props;
  const {isFocused} = state;
  return {
    $isFocused: isFocused,
    $disabled: disabled,
    $error: error,
    $positive: positive,
    $adjoined: adjoined,
    $size: size,
    $required: required,
  };
}
