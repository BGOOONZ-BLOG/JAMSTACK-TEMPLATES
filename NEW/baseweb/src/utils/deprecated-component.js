/*
Copyright (c) 2018-2020 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import * as React from 'react';

// eslint-disable-next-line flowtype/no-weak-types
export default function(Component: any, displayName: string) {
  // eslint-disable-next-line flowtype/no-weak-types
  return React.forwardRef<any, any>((props, ref) => {
    if (__DEV__) {
      console.warn(
        `We have stabilized the ${displayName} component, so you can drop the "Unstable_" prefix from your imports. We will remove the "Unstable_" exports soon, so please make these changes as soon as possible!`,
      );
    }
    return <Component {...props} ref={ref} />;
  });
}
