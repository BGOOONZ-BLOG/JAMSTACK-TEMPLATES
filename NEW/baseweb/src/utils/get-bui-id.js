/*
Copyright (c) 2018-2020 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

/**
 * For aria and other attributes like htmlFor, we often need
 * to attach identifiers to elements. Making `id` a required
 * prop in our components shifts this burden to our customers
 * but it's nice to try to automatically provide these unique
 * ids when possible.
 *
 * Generating unique IDs that match across server/client renders
 * is a hard problem that doesn't have any perfect solutions
 * currently [1] [2]. UUIDs/Math.random will never match, whereas
 * auto-incrementing IDs may work if server and client render
 * exactly the same DOM, so that's what we'll use here. If this
 * is insufficient for users they can apply ids manually.
 *
 * One other technique that we can use internally is to do a
 * setState in componentDidMount or something similar, so that
 * these ids are only used client-side, but this could of course
 * have perf implications.
 *
 * [1] https://github.com/facebook/react/issues/5867
 * [2] https://github.com/reactjs/rfcs/pull/32/files
 */
let id = 0;

export default function getBuiId() {
  return `bui-${++id}`;
}
