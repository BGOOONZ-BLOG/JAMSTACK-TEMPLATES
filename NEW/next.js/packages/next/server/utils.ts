import Observable from 'next/dist/compiled/zen-observable'
import { BLOCKED_PAGES } from '../shared/lib/constants'

export function isBlockedPage(pathname: string): boolean {
  return BLOCKED_PAGES.includes(pathname)
}

export function cleanAmpPath(pathname: string): string {
  if (pathname.match(/\?amp=(y|yes|true|1)/)) {
    pathname = pathname.replace(/\?amp=(y|yes|true|1)&?/, '?')
  }
  if (pathname.match(/&amp=(y|yes|true|1)/)) {
    pathname = pathname.replace(/&amp=(y|yes|true|1)/, '')
  }
  pathname = pathname.replace(/\?$/, '')
  return pathname
}

export type RenderResult = Observable<string>

export function mergeResults(results: Array<RenderResult>): RenderResult {
  // @ts-ignore
  return Observable.prototype.concat.call(...results)
}

export async function resultsToString(
  results: Array<RenderResult>
): Promise<string> {
  const chunks: string[] = []
  await mergeResults(results).forEach((chunk: string) => {
    chunks.push(chunk)
  })
  return chunks.join('')
}
