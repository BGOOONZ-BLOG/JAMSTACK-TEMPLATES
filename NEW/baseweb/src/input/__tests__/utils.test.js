/*
Copyright (c) 2018-2020 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import {getSharedProps} from '../utils.js';
import {ADJOINED, SIZE} from '../index.js';

test('Utils - getSharedProps', () => {
  const props = {
    id: 'id',
    inputRef: {current: {focus: () => {}}},
    adjoined: ADJOINED.none,
    size: SIZE.default,
    required: false,
    disabled: false,
    error: false,
    positive: false,
    value: 'value',
    placeholder: 'placeholder',
    onChange: jest.fn(),
    autoFocus: false,
    isFocused: false,
  };
  const state = {
    isFocused: true,
  };
  // $FlowFixMe
  expect(getSharedProps(props, state)).toMatchSnapshot(
    'getSharedProps returns correct object',
  );
});
