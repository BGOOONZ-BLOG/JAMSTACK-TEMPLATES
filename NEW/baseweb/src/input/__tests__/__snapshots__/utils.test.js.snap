// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils - getSharedProps: getSharedProps returns correct object 1`] = `
Object {
  "$adjoined": "none",
  "$disabled": false,
  "$error": false,
  "$isFocused": true,
  "$positive": false,
  "$required": false,
  "$size": "default",
}
`;
